File: src/fraction.cpp
#include "fraction.h"

void In(fraction &num, std::ifstream &in) {
    in >> num.a >> num.b;
}

void InRnd(fraction &num) {
    num.a = rand() % 1000;
    num.b = rand() % 1000 + 1; 
}

// Вывод в поток
void Out(fraction &num, std::ofstream &out) {
    out << num.a << " / " << num.b << " ";
}

double RationalRepr(fraction &num) {
    return (double) num.a / (double) num.b;
}

File: hw3/src/fraction.py

class fraction:
    """
        Fraction
        Num = a / b
    """
    def __init__(self, a, b):
        if b < 0:
            b *= -1
            a *= -1

        if b == 0:
            raise ZeroDivisionError()

        self.a = int(a)
        self.b = int(b)

    def rational_repr(self):
        return float(self.a) / float(self.b)

    def __str__(self):
        return f"{self.a} / {self.b}"
